sqlitecluster/SQLiteNode.h

```diff
@@ -175,32 +195,45 @@ class SQLiteNode : public STCPNode {
     //
     // _cleanCommand() is called when the command is closed; use it for any
     // final cleanup operations.
-    //
-    virtual bool _peekCommand(SQLite& db, Command* command) = 0;
-    virtual void _processCommand(SQLite& db, Command* command) = 0;
     virtual void _abortCommand(SQLite& db, Command* command) = 0;
     virtual void _cleanCommand(Command* command) = 0;
+    // This returns `true` if we need to commit something to the database after this operation, false otherwise. The
+    // two common cases for why this would return false are that the command threw an exception, or it succeeded but
+    // didn't need to write anything to the database.
+    virtual bool _processCommand(SQLite& db, Command* command) = 0;
+
+    // Should return true if an external queue accepted the command. If so, this node now keeps no reference to the
+    // command, and the external queue is responsible for calling `reopenCommand` to return the command to this node.
+    virtual bool _passToExternalQueue(Command* command) { return false; };
 
     // Wrappers for peek and process command to keep track of processing time.
     bool _peekCommandWrapper(SQLite& db, Command* command);
-    void _processCommandWrapper(SQLite& db, Command* command);
+    bool _processCommandWrapper(SQLite& db, Command* command);
```

Comment:
@tylerkaraszewski  - Where does the new boolean return value of `_processCommand()` or `_processCommandWrapper()` get used?  I don't see it anywhere in this PR.